File: backend/src/main/java/com/coing/domain/bookmark/controller/dto/BookmarkResponse.java
package com.coing.domain.bookmark.controller.dto;

import com.coing.domain.bookmark.entity.Bookmark;

public record BookmarkResponse(
	String code,
	String koreanName,
	String englishName,
	Boolean isBookmarked
) {
	public static BookmarkResponse of(Bookmark bookmark) {
		return new BookmarkResponse(
			bookmark.getMarket().getCode(),
			bookmark.getMarket().getKoreanName(),
			bookmark.getMarket().getEnglishName(),
			true
		);
	}
}

File: backend/src/main/java/com/coing/domain/bookmark/controller/dto/BookmarkResponse.kt
package com.coing.domain.bookmark.controller.dto

import com.coing.domain.bookmark.entity.Bookmark

data class BookmarkResponse(
    val code: String,
    val koreanName: String,
    val englishName: String,
    val isBookmarked: Boolean
) {
    companion object {
        fun of(bookmark: Bookmark): BookmarkResponse {
            return BookmarkResponse(
                code = bookmark.market.code,
                koreanName = bookmark.market.koreanName,
                englishName = bookmark.market.englishName,
                isBookmarked = true
            )
        }
    }
}
